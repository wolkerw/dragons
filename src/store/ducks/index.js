import { combineReducers } from "redux";

import login from "./login";
import input from "./login";
import dragons from "./dragons";
import dragonList from "./dragonList";

export default combineReducers({
    login,
    dragons,
    dragonList,
    input
});
