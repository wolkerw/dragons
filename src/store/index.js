import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { composeWithDevTools } from 'redux-devtools-extension'
import reducers from './ducks';
import rootSaga from "./sagas";

/**
 * Indice dos Ducks e Sagas
 * @author FÃ¡bio Sartori
 * @copyright 06/2019
 */
const sagaMiddleware = createSagaMiddleware();

let midCompose = "";

if (process.env.NODE_ENV === "development") {
    midCompose = composeWithDevTools(
        applyMiddleware(sagaMiddleware)
        );
    } else {
    midCompose = applyMiddleware(sagaMiddleware);
}

const store = createStore(reducers, midCompose);

sagaMiddleware.run(rootSaga);

export default store;