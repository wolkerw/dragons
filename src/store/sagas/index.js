import { all } from "redux-saga/effects";
// import { getFreeGatesWatcher } from "./freeGates";
import { getOrdersWatcher } from "./orders";
import { getAuthenticationWatcher } from "./authentication";
// import { getAuthorizationWatcher } from "./authorization";
// import { getReconcileWatcher } from "./reconcile";
//import { getReport } from "./relatorio";

export default function* rootSaga() {
    yield all([
        // getFreeGatesWatcher(),
        getOrdersWatcher(),
        getAuthenticationWatcher()
        // getAuthorizationWatcher(),
        // getReconcileWatcher()
        //getReport()
    ]);
}
