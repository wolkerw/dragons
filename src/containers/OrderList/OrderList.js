import React, { Component, Fragment } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import PropTypes from "prop-types";
import { Container } from "react-bootstrap";
// import debounce from "lodash.debounce";
import { Creators as List } from "../../store/ducks/orderList";
import Order from "../../components/Order/Order";
import "./OrderList.scss";

class OrderList extends Component {
    constructor(props) {
        super(props);

        // Initial State
        this.state = {
            error: false,
            // hasMore: true,
            isLoading: false,
            list: []
        };

        // this.pageScroll();
    }

    componentDidMount() {
        // Fazendo o primeiro load de usuários
        this.props.getOrders();
        this.props.orderList.cleanOrders();

        this.loadList();
    }

    // pageScroll() {
    //     // Configurando debounce e evento de scroll
    //     window.onscroll = debounce(() => {
    //         const {
    //             loadList,
    //             state: { error, isLoading /*, hasMore*/ }
    //         } = this;

    //         //Evento para interromper o loading
    //         if (error || isLoading /* || !hasMore*/) return;

    //         // Verificando se o scroll chegou a base da tela
    //         if (
    //             window.innerHeight + document.documentElement.scrollTop ===
    //             document.documentElement.offsetHeight
    //         ) {
    //             loadList();
    //         }
    //     }, 100);
    // }

    handleChecked = (checked, order) => {
        if (!checked) {
            this.props.orderList.addOrder(order);
        } else {
            this.props.orderList.removeOrder(order);
        }
    };

    componentDidUpdate() {
        if (
            this.props.orders &&
            this.props.orders.length &&
            this.props.orders[0] === "error" &&
            !this.state.error
        ) {
            this.setState({
                error: "Erro1",
                isLoading: false
            });
        } else if (
            this.props.orders &&
            this.props.orders.length &&
            !this.state.list.length
        ) {
            this.setState({
                list: this.props.orders,
                isLoading: false
            });

            //return result;
            //Populando o array de pedidos
            /*const nextOrder = this.state.orders.map(order => order);

            // Adiciona os próximos pedidos ao state
            this.setState({
                //Limitando o número total de registros
                hasMore: this.state.list.length < 0,
                isLoading: false,
                list: [...this.state.list, ...nextOrder]
            });*/
        }
    }

    loadList = () => {
        this.setState({ isLoading: true });
        this.props.getOrders();
    };

    removeDragon = id => {
        this.props.removeDragon(id);

        this.setState({
            list: this.state.list.filter(function(item) {
                return item.id !== id;
            })
        });
    };

    editDragon = id => {
        this.props.history.push(`/dragon/${id}`);
    };

    render() {
        const { error, /*hasMore,*/ isLoading, list } = this.state;

        //const { checkList } = this.props;

        //const header = ["Nome", "Tipo", "Remover", "Editar"];

        //let cols = 12;
        //const length = header.length;

        //cols = checkList === true ? cols - 1 : cols;

        //const math = Number((cols / length).toFixed(1));

        //const size = math > 1 ? math : 1;

        //const style = ("col-" + size).replace(".", "-");

        return (
            <Container className="text-center order-list-container">
                <div className="row order-list-header">
                    <div className="col-4" key="Nome">
                        <span>Nome</span>
                    </div>
                    <div className="col-4" key="Tipo">
                        <span>Tipo</span>
                    </div>
                    <div className="col-2" key="Remover">
                        <span>Remover</span>
                    </div>
                    <div className="col-2" key="Editar">
                        <span>Editar</span>
                    </div>
                </div>
                {!error &&
                    !isLoading &&
                    list.map((order, i) => {
                        return (
                            <Fragment key={i}>
                                <Order
                                    data={order}
                                    functionProps={this.handleChecked}
                                    id={`order-list-order-checkbox-${i + 1}`}
                                    status="idStatusConciliacao"
                                    removeDragon={this.removeDragon}
                                    editDragon={this.editDragon}
                                    checkList
                                />
                            </Fragment>
                        );
                    })}
                {error && (
                    <div>
                        Erro ao buscar os dragões. Verifique se tem conexão com
                        aa internet e tente novamente mais tarde.
                    </div>
                )}
                {isLoading && (
                    <div>
                        <p className="text-3 text-color-primary my-0 pt-1">
                            Carregando...
                        </p>
                    </div>
                )}
                {/* {!hasMore && (
                    <div>
                        <p className="text-3 text-color-primary font-weight-bolder my-3 pt-1">
                            Não há mais dados
                        </p>
                    </div>
                )} */}
            </Container>
        );
    }
}

const mapStateToProps = state => {
    return {
        orderList: state.orderList,
        orders: state.orders
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getOrders: () => dispatch({ type: "GET_ORDERS" }),
        removeDragon: id =>
            dispatch({
                type: "REMOVE_DRAGON",
                id: id
            }),
        orderList: bindActionCreators(List, dispatch)
    };
};

OrderList.propTypes = {
    checkList: PropTypes.bool
};

export const OrderListWithoutWithRouter = connect(
    mapStateToProps,
    mapDispatchToProps
)(OrderList);

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(OrderList)
);
