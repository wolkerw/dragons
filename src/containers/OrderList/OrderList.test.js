import React from "react";
import { OrderListWithoutWithRouter } from "./OrderList";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
// import messages from "../../assets/i18n";
import { Provider } from "react-redux";
import store from "./../../store";

configure({ adapter: new Adapter() });

const country = "pt-BR";

describe("OrderList Container", () => {
    const component = (
        <IntlProvider
            locale={country}
            // key={country}
            // messages={messages[{ country }]}
            // defaultLocale={country}
        >
            <Provider store={store}>
                <OrderListWithoutWithRouter checkList />
            </Provider>
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("generated an order list with a header inside", () => {
        const wrapper = mount(component);
        const header = wrapper.find(".order-list-header");
        expect(header.length).toEqual(1);
    });

    it("generated an order list that is loading orders", () => {
        const wrapper = mount(component);
        const loading = wrapper
            .find(".container div")
            .last()
            .text();
        expect(loading).toBe("Carregando...");
    });
});
