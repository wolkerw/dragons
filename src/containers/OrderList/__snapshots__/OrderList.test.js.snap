// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList Container matches the snapshot 1`] = `
<div
  className="text-center order-list-container container"
>
  <div
    className="row order-list-header"
  >
    <div
      className="col-4"
    >
      <span>
        Nome
      </span>
    </div>
    <div
      className="col-4"
    >
      <span>
        Tipo
      </span>
    </div>
    <div
      className="col-2"
    >
      <span>
        Remover
      </span>
    </div>
    <div
      className="col-2"
    >
      <span>
        Editar
      </span>
    </div>
  </div>
  <div>
    <p
      className="text-3 text-color-primary my-0 pt-1"
    >
      Carregando...
    </p>
  </div>
</div>
`;
